rpcs3/rpcs3qt/main_window.cpp

```diff
@@ -912,11 +938,64 @@ void main_window::DecryptSPRXLibraries()
 				{
 					gui_log.error("Failed to create %s", new_path);
 				}
+
+				break;
 			}
-			else
+			else if (!tried)
 			{
-				gui_log.error("Failed to decrypt %s", old_path);
+				// Allow the user to manually type KLIC if decryption failed
+
+				const std::string filename = old_path.substr(old_path.find_last_of(fs::delim) + 1);
+
+				input_dialog dlg(32, QString{}, tr("Enter KLIC of %0").arg(qstr(filename)),
+					tr("Hexadecimal only."), "00000000000000000000000000000000", this);
+
+				QFont mono = QFontDatabase::systemFont(QFontDatabase::FixedFont);
+				mono.setPointSize(8);
+				dlg.edit_input_font(mono);
+				dlg.disable_clear_button();
+				dlg.set_validator(new QRegExpValidator(QRegExp("^[a-fA-F0-9]*$"))); // HEX only
+
+				QString result;
+				connect(&dlg, &input_dialog::text_changed, [&](const QString& text)
+				{
+					result = text;
+				});
+
+				if (dlg.exec() == QDialog::Accepted)
+				{
+					const std::string text = sstr(result);
+
+					if (text.size() == 32)
+					{
+						u32 index = 0;
+
+						for (auto& ch : klic._bytes)
+						{
+							auto res = std::from_chars(&text[index], &text[index] + 2, ch, 16);
+
+							if (res.ec != std::errc() || res.ptr != &text[index] + 2)
+							{
+								break;
+							}
+
+							index += 2;
+						}
+
+						// Retry with specified KLIC
+						if (index == 32)
+						{
+							tried = true;
+							klics.emplace_back(klic);
```

Comment:
Doesn't really make sense to try all the klics again for this file (log spam and wasted efford)

Reply:
Fixed.